substrate/frame/bags-list/src/lib.rs

```diff
@@ -301,11 +338,20 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::rebag_non_terminal().max(T::WeightInfo::rebag_terminal()))]
 		pub fn rebag(origin: OriginFor<T>, dislocated: AccountIdLookupOf<T>) -> DispatchResult {
 			ensure_signed(origin)?;
+
+			ensure!(LockOrder::<T, I>::get().is_none(), Error::<T, I>::ReorderingLocked);
+
 			let dislocated = T::Lookup::lookup(dislocated)?;
 			let current_score = T::ScoreProvider::score(&dislocated);
-			let _ = Pallet::<T, I>::do_rebag(&dislocated, current_score)
-				.map_err::<Error<T, I>, _>(Into::into)?;
-			Ok(())
+
+			if Self::remove_if_zero(&dislocated)? {
+				// removed `dislocated.
+				Ok(())
+			} else {
+				Pallet::<T, I>::do_rebag(&dislocated, current_score)
+					.map_err::<Error<T, I>, _>(Into::into)?;
+				Ok(())
+			}
```

Comment:
can be simplified a bit
```suggestion

			if Self::remove_if_zero(&dislocated)? {
				// removed `dislocated.
				return Ok(())
			} 
			
			Pallet::<T, I>::do_rebag(&dislocated, current_score)
				.map_err::<Error<T, I>, _>(Into::into)
```